HowTo/use_system-assigned_managed_identities.md

```diff
@@ -0,0 +1,90 @@
+
+## STEP 0 create your resources
+
+> **Note:** If you already have resources you want to use, skip to step 1
+
+For this example we will need to create the following resources:
+  
+- Web App
+- App Service Plan
+- Azure Container Registry
+
+``` bash
+# Modify for your environment
+
+location = "west us"    #Azure Region where the resources will be created
+RG_Name = "RG-test"     #Name of the resource group that will contain the resources
+ASP_Name = "ASP-Test"   #Name of the App Service plan
+Web_Name = "WebApp-101" #Globally unique name of the WebApp
+ACR_Name = "bkACRtest"  #Name of the Azure Container Registry
+
+az group create -n $RG_Name -l $location -o none
+az acr create -n $ACR_Name --sku standard -g $RG_Name -l $location -o none
+az appservice plan create --is-linux -n $ASP_Name --sku p1v2 -g $RG_Name -l $location -o none
+az webapp create -n $Web_Name -g $RG_Name -p $ASP_Name -i "nginx" -o none
+
+az resource list -g $RG_Name -o table
+```
+
+>**Note:** If you are starting with an empty ACR instance, you will need to [push an image into ACR](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-docker-cli) so that you can later configure you app to use it.
+
+## STEP 1: Assign an identity to your WebApp
+
+This step will configure the webapp to use a system-assigned identity. System-assigned identities are created on the fly and unlike 'service-principle' they don't require you to be a subscription level admin to create and use them.
+
+```bash
+# Modify for your environment
+
+#Assign identity to the app
+az webapp identity assign -g $RG_Name -n $Web_Name -o none
+
+#Collect id for the system-assigned identity
+Identity_ID = $(az webapp show -g $RG_Name -n $Web_Name --query identity.principalId --output tsv)
+```
+
+## STEP 2: Grant access to the identity on ACR
+
+This step will register the identity with ACR and grant it the minimum permission necessary for a webapp to pull and host containers from it.
+
+``` bash
+# Modify for your environment
+
+#Resource ID for ACR instance create in STEP 0
+ACR_ID=$(az acr show -g $RG_Name -n $ACR_Name --query id --output tsv)
+
+#ACR will allow the identity to perform pull operations and nothing more
+az role assignment create --assignee $Identity_ID --scope $ACR_ID --role acrpull -o none
+```
+
+## Step 3: Configure WebApp configuration
+
+This step will configure the webapp to use the assigned identity for docker pull operations and update the configuration to point to ACR and the Image:Tag for the container you want to use.
+
+```bash
+# Modify for your environment
+
+#URL for ACR instance create in STEP 0
+ACR_URL=$(az acr show -g $RG_Name --n $ACR_Name --query loginServer --output tsv)
+
+#Resource ID for WebApp configuration.
+Webapp_Config = $(az webapp show -g $RG_Name -n $Web_Name --query id --output tsv) + "/config/web"
+
+Image ="myapp:latest" #Docker image and tag to pull
+
+FX_Version = '"Docker|"' + $ACR_URL + "/" + $Image
```

Comment:
Do we need the double quotes around Docker| here?

Reply:
Fixed this. I was originally writing and testing this on my windows laptop + powershell console, switched over to mac + zsh and cleanup the script